src/tests/unit/tests/common/components/issue-filing-button.test.tsx

```diff
@@ -0,0 +1,133 @@
+            createBugFilingUrl: () => 'test url',
+        };
+        userConfigurationStoreData = {
+            bugService: testKey,
+            bugServicePropertiesMap: {},
+        } as UserConfigurationStoreData;
+        environmentInfoProviderMock = Mock.ofType(EnvironmentInfoProvider);
+        bugFilingServiceProviderMock = Mock.ofType(BugFilingServiceProvider);
+        bugActionMessageCreatorMock = Mock.ofType(BugActionMessageCreator);
+        environmentInfoProviderMock
+            .setup(envp => envp.getEnvironmentInfo())
+            .returns(() => {
+                return {
+                    extensionVersion: '1',
+                    axeCoreVersion: '2',
+                    browserSpec: 'spec',
+                };
+            })
+            .verifiable();
+        bugFilingServiceProviderMock
+            .setup(bp => bp.forKey(testKey))
+            .returns(() => testBugService)
+            .verifiable();
+    });
+
+    test.each([true, false])('render: isSettingsValid: %s', isSettingsValid => {
+        testBugService.isSettingsValid = () => isSettingsValid;
+        const props: IssueFilingButtonProps = {
+            deps: {
+                bugActionMessageCreator: bugActionMessageCreatorMock.object,
+                environmentInfoProvider: environmentInfoProviderMock.object,
+                bugFilingServiceProvider: bugFilingServiceProviderMock.object,
+            } as IssueFilingButtonDeps,
+            issueDetailsData: {
+                pageTitle: 'pageTitle',
+                pageUrl: 'http://pageUrl',
+                ruleResult: null,
+            },
+            userConfigurationStoreData: userConfigurationStoreData,
+        };
+        const wrapper = shallow(<IssueFilingButton {...props} />);
+        expect(wrapper.debug()).toMatchSnapshot();
+
+        environmentInfoProviderMock.verifyAll();
+        bugFilingServiceProviderMock.verifyAll();
+    });
+
+    test('onclick: valid settings, file bug', () => {
+        bugActionMessageCreatorMock
+            .setup(messageCreator => messageCreator.trackFileIssueClick(It.isAny(), testKey as any))
+            .verifiable(Times.once());
+        const props: IssueFilingButtonProps = {
+            deps: {
+                bugActionMessageCreator: bugActionMessageCreatorMock.object,
+                environmentInfoProvider: environmentInfoProviderMock.object,
+                bugFilingServiceProvider: bugFilingServiceProviderMock.object,
+            } as IssueFilingButtonDeps,
+            issueDetailsData: {
+                pageTitle: 'pageTitle',
+                pageUrl: 'http://pageUrl',
+                ruleResult: null,
+            },
+            userConfigurationStoreData: userConfigurationStoreData,
+        };
+        const wrapper = shallow(<IssueFilingButton {...props} />);
+
+        wrapper.find(DefaultButton).simulate('click');
+
+        environmentInfoProviderMock.verifyAll();
```

Comment:
Why are we verifying `environmentInfoProviderMock` and `bugFilingServiceProviderMock`? the function being tested here (`trackFileIssueClick`) doesn't directly use any of this deps. Are we over verifying things in this test? Things that we are also verifying on a different test - one with a better context about the usage of the 2 mentioned mocks?

Reply:
right, it is not necessary.